src/initgeneration/initgeneration.go

```diff
@@ -88,26 +88,22 @@ func (g *InitGenerator) GenerateForDynakube(ctx context.Context, dk *dynatracev1
 
 	coreLabels := kubeobjects.NewCoreLabels(dk.Name, kubeobjects.WebhookComponentLabel)
 	secretQuery := kubeobjects.NewSecretQuery(ctx, g.client, g.apiReader, log)
+	secret := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   config.AgentInitSecretName,
+			Labels: coreLabels.BuildMatchLabels(),
+		},
+		Data: data,
+		Type: corev1.SecretTypeOpaque,
+	}
 
-	for _, targetNs := range nsList {
-		secret := &corev1.Secret{
-			TypeMeta: metav1.TypeMeta{},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      config.AgentInitSecretName,
-				Namespace: targetNs.Name,
-				Labels:    coreLabels.BuildMatchLabels(),
-			},
-			Data: data,
-			Type: corev1.SecretTypeOpaque,
-		}
-
-		err = secretQuery.CreateOrUpdate(*secret)
-		if err != nil {
-			return errors.WithStack(err)
-		}
+	updateCnt, err := secretQuery.CreateOrUpdateForNamespacesList(secret, nsList)
+	if err != nil {
+		return err
 	}
 
-	log.Info("done updating init secrets")
+	log.Info("done updating init secrets", "update/creation count", updateCnt)
```

Comment:
log could be once in `CreateOrUpdateForNamespacesList` using the `secret.Name` instead of having it after every call of the function

Reply:
The reason why I did not do that was that for me `SecretQuery` is just a utility and should not force its user to actually produce logs. This is already a pain when initially starting up on a multiple namespace cluster. 

It will give you lot's of log lines "created secret" because `SecretQuery` has a log coded into it. - In my opinion on the long run we should remove the logs from our kubernetes utility functions all together.

Reply:
having the `create` and `update` logs make sense most of the time, but I agree when you are bulk creating/updating then it gets a bit too much.

having reasonable logs handled by the query makes sense.

you could have a 
```
func (query SecretQuery) create(secret corev1.Secret) error {...)
```
that does not log, so it can be used by the other `query` functions in a way that it only logs what is necessary

Reply:
Like the idea of the private function, added it